app/views/NotificationPreferencesView/index.tsx

```diff
@@ -22,18 +23,33 @@ const styles = StyleSheet.create({
 	}
 });
 
-class NotificationPreferencesView extends React.Component {
+interface INotificationPreferencesView {
+	navigation: StackNavigationProp<any, 'NotificationPreferencesView'>;
+	route: RouteProp<
+		{
+			NotificationPreferencesView: {
+				rid: string;
+				room: {
+					observe: Function;
+				};
+			};
+		},
+		'NotificationPreferencesView'
+	>;
+	theme: string;
+}
+
+class NotificationPreferencesView extends React.Component<INotificationPreferencesView, any> {
 	static navigationOptions = () => ({
 		title: I18n.t('Notification_Preferences')
 	});
 
-	static propTypes = {
-		navigation: PropTypes.object,
-		route: PropTypes.object,
-		theme: PropTypes.string
-	};
+	private mounted: boolean;
+	private rid: string | undefined;
+	private roomObservable: any;
+	private subscription: any;
```

Comment:
Done! But, I didn't create as `undefined`... I preferred create as not required... because we have a if to check if exist a `room` and a `room.observe()`